networkit/test/test_distance.py

```diff
@@ -188,11 +229,49 @@ def testMultiTargetSTSP(self):
 					algo = nk.distance.MultiTargetDijkstra(g, source, targets)
 				else:
 					algo = nk.distance.MultiTargetBFS(g, source, targets)
-				algo.setTargets(targets)
+				algo.setSource(source)
+				algo.setTarget(targets[nTargets-1])
+				algo.run()
+				self.assertLessEqual(len(algo.getPredecessors()),g.numberOfNodes())
+
+	def testSSSP(self):
+		for g in self.genERGraphs():
+			source = nk.graphtools.randomNode(g)
+			for nTargets in [1, 10, 50]:
+				targets = nk.graphtools.randomNodes(g, nTargets)
+				algo = None
+				if g.isWeighted():
+					algo = nk.distance.Dijkstra(g, source, targets, storeNodesSortedByDistance = True)
+				else:
+					algo = nk.distance.BFS(g, source, targets)
 				algo.setSource(source)	
+				algo.setTarget(targets[nTargets-1])
 				algo.run()
-				self.assertLessEqual(len(algo.getPredecessors()), g.numberOfNodes())	
-				self.assertEqual(len(algo.getDistances()), len(targets))
+				self.assertLessEqual(len(algo.getPredecessors(source)),g.numberOfNodes())
+				self.assertLessEqual(len(targets),len(algo.getDistances()))
+				self.assertLessEqual(len(algo.getPaths(targets[nTargets-1])),g.numberOfNodes())
+				if g.isWeighted():
+					self.assertLessEqual(len(algo.getNodesSortedByDistance()), g.numberOfNodes())
+				self.assertLessEqual(1.0, algo.numberOfPaths(targets[nTargets-1]))
+
+	def testDynSSSP(self):
+		for g in self.genERGraphs():
+			source = nk.graphtools.randomNode(g)
+			for nTargets in [1, 10, 50]:
+				targets = nk.graphtools.randomNodes(g, nTargets)
+				algoDyn = None
+				if g.isWeighted():
+					algoDyn = nk.distance.DynDijkstra(g, source)
+				else:
+					algoDyn = nk.distance.DynBFS(g, source)
+				algoDyn.run()
+				if not g.isWeighted():
+					batch1 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 3, 7, 1.0)
+					batch2 = nk.dynamics.GraphEvent(nk.dynamics.GraphEventType.EDGE_ADDITION, 1, 5, 1.0)
+					batch = [batch1,batch2]
+					algoDyn.updateBatch(batch)
```

Comment:
Here, you check only for `updateBatch` - is `update` already tested?

Reply:
update() had been used in testDynPrunedLandmarkLabeling, therefore, i decided to only test updateBatch()